src/kernels/gpu_reference_kernel/naive_conv.cpp

```diff
@@ -114,10 +114,120 @@ inline __device__ __host__ int8_t cast_to(const int32_t& val)
+
+    __host__ __device__ MyArray()                   = default;
+    __host__ __device__ MyArray(const MyArray&)     = default;
+    __host__ __device__ MyArray(MyArray&&) noexcept = default;
+    __host__ __device__ MyArray& operator=(const MyArray&) = default;
+    __host__ __device__ MyArray& operator=(MyArray&&) noexcept = default;
+    __host__ __device__ ~MyArray()                             = default;
+};
+
+using StrideIndexType = size_t;
+using Strides5D       = MyArray<StrideIndexType, 5u>;
+using Strides6D       = MyArray<StrideIndexType, 6u>;
+#else
+
+extern "C" typedef int StrideIndexType;
+
+extern "C" typedef struct
+{
+    StrideIndexType v[5];
+} Strides5D;
+
+extern "C" typedef struct
+{
+    StrideIndexType v[6];
+} Strides6D;
+
+extern "C" __global__ void testKernel(void* ptr_a,
+                                      void* ptr_b,
+                                      void* ptr_c,
+                                      Strides5D in_strides,
+                                      Strides5D wei_strides,
+                                      Strides5D out_strides)
+{
+
+    if(blockIdx.x == 0 && threadIdx.x == 0)
+    {
+        printf("sizeof(Strides5D) = %lu\n", sizeof(Strides5D));
+        printf("%p, %p, %p, %p\n", &ptr_a, &ptr_b, &ptr_c, &in_strides);
+        printf("in_strides = [");
+        for(int i = 0; i < 5; ++i)
+        {
+            // printf("%d,", in_strides.v[i]);
+            printf("%d,", in_strides[i]);
+        }
+        printf("]\n");
+        printf("wei_strides = [");
+        for(int i = 0; i < 5; ++i)
+        {
+            // printf("%d,", wei_strides.v[i]);
+            printf("%d,", wei_strides[i]);
+        }
+        printf("]\n");
+        printf("out_strides = [");
+        for(int i = 0; i < 5; ++i)
+        {
+            // printf("%d,", out_strides.v[i]);
+            printf("%d,", out_strides[i]);
+        }
+        printf("]\n");
+    }
+}
+#endif
+
+template <typename StrideArray>
+__device__ void printStrideArray(const char* name, const StrideArray& sarr)
+{
+    printf("%s = [", name);
+    for(int i = 0; i < StrideArray::SIZE; ++i)
+    {
+        printf("%d,", sarr[i]);
+    }
+    printf("]\n");
+}
+
+// TODO(Amber): remove template parameter 'bool ASSUME_PACKED' in a follow up PR
+// Notes (Amber):
+// * The following code used to assume that group (G) is an implicit
+// dimension, i.e. c= c_per_group * group and k = k_per_group * group. This is not
+// true for non-packed case because group (G) dimension needs to have its stride
+// explicitly specified for address math to make sense. This is also how
+// composable_kernel (CK) treats G dimension. Which is why nchw should be ngchw,
+// and nhwc should be nhwgc. Same follows for the 3D case.
+// * strides here are in the little-endian order, i.e., for NHWC, stride for N is
+// at index 3 while stride for C is at index 0. This is reverse of how strides are
+// stored in tensor descriptors, which are big-endian.
```

Comment:
[R] The term "endianness" is not applicable here. You can simple say that strides are in NHWC layout.

Reply:
I disagree respectfully. `endianness` matters. I can store the strides for NHWC in an array left-to-right or right-to-left. Currently MIOpen makes an even weirder choice of storing all strides in `NCHW` order left-to-right. We should discuss this in the near future.

Reply:
@amberhassaan Then I recommend to invent a new word. "_Weirdness_" seems to be a good candidate ;) In the computing world, endianness has very specific meaning.

> Currently MIOpen makes an even weirder choice of storing all strides in `NCHW` order left-to-right.

This is sad. Fortunately, I am not involved in such design. I don’t have time to review everything. Although this is for good - I don’t want to end my days in Bedlam ;)

Reply:
@averinevg Can you please look at this review thread as well.

Reply:
@amberhassaan The concept of `endianness` applies to byte order in integers. The order of variables in a sequence is ... just the order of variables in a sequence :smile:

Reply:
The concept I'm trying to describe is very very close to endianness and I believe it's appropriate to borrow this term. See for example: https://hackage.haskell.org/package/polynomial-0.5/docs/Math-Polynomial.html The question in this link is how to store polynomials, and I think big-endian vs little-endian conveys the concept clearly as opposed to saying "left-to-right" or "right-to-left" because there is no left or right in memory and the terms left or right are not clear.

Reply:
@amberhassaan The term `endianness` has a very specific definition in the technical literature. Any developer can contribute to this library, and reading your comment they will understand it not quite the way you intended. Therefore, in order not to cause confusion, let's use generally accepted terms in accordance with their meaning. I think that references to the terms `big endian` and `little endian` can be removed from your comment; the meaning will not be affected and there will be no misunderstandings.

Reply:
next commit will remove the terms `endianness`.

Reply:
@amberhassaan [notice] I think you meant next PR.